src/crypto/tls/handshake_client_tls13.go

```diff
@@ -709,6 +709,30 @@ func certificateRequestInfo(certReq *certificateRequestMsgTLS13, vers uint16) *C
 	return cri
 }
 
+func getClientDelegatedCredential(cri *CertificateRequestInfo, cert *Certificate) (*DelegatedCredentialStruct, error) {
+	if len(cert.DelegatedCredentials) == 0 {
+		return nil, errors.New("No Delegated Credential found.")
+	}
+
+	if len(cert.DelegatedCredentials) == 1 {
+		// There's only one choice, so no point doing any work.
+		return &cert.DelegatedCredentials[0], nil
+	}
```

Comment:
What happens if the client uses a DC whose algorithm isn't supported by the server?

Reply:
Good question! So, these two functions `getDelegatedCredential` and `getClientDelegatedCredential` emulate `getCertificate` and `getClientCertificate`: their aim is to return one credential/certificate even if it is not "valid" (I think the logic in Golang was that these functions should just return a certificate).  In the actual handshake, the check will be done by selecting the signature algorithm: https://github.com/cloudflare/go/blob/dc-api/src/crypto/tls/handshake_server_tls13.go#L522 If it is unsupported, it will error.

Reply:
Ah, right you are. IMO it would be cleaner to return `(nil, nil)` if no compatible DC is found. This should have the same affect, since the branch here wouldn't be taken: https://github.com/cloudflare/go/blob/dc-api/src/crypto/tls/handshake_server_tls13.go#L514

I think this is better because the caller is guaranteed that *if* a DC is returned, then it can be used in the handshake. Alternatively, I would suggest adding a comment to `get[Client]DelegatedCredential` noting that the returned DC may not be suitable for the handshake.

Reply:
Addressed now.